src/mem/heap/heap_simple.c

```diff
@@ -93,10 +93,15 @@ void free(void *ptr) {
 }
 
 void *malloc(size_t size) {
+	return memalign(sizeof(void*), size);
+}
+
+void *memalign(size_t boundary, size_t size) {
 	void *ptr;
 	struct mem_control_block *mcb, *next_mcb;
+	size_t shift;
 
-	if (size == 0) {
+	if (size == 0 || (boundary & (boundary - 1)) != 0 || boundary % sizeof(void*) != 0) {
```

Comment:
I think we can remove `boundary % sizeof(void*) != 0` now, because this aligment reqired only for malloc()

Reply:
Done.